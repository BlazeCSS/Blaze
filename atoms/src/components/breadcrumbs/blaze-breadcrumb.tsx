import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'blaze-breadcrumb'
})
export class Breadcrumb {

  @Prop() href: string;

  render() {
    return (
      <li class="c-breadcrumbs__crumb">
        {
          this.href ?
          <a class="c-link" href={this.href}>
            <slot />
          </a>
          :
          <slot />
        }
      </li>
    );
  }
}
