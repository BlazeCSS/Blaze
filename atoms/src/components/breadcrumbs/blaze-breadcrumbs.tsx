import { Component, Element } from '@stencil/core';

@Component({
  tag: 'blaze-breadcrumbs'
})
export class Breadcrumbs {

  @Element() el: HTMLElement;

  componentDidLoad() {
    const lastCrumb = this.el.querySelector('blaze-breadcrumb:last-child .c-breadcrumbs__crumb');

    if (lastCrumb) {
      lastCrumb.classList.add('c-breadcrumb__crumb--last');
    }
  }

  render() {
    return (
      <ol class="c-text c-breadcrumbs">
        <slot />
      </ol>
    );
  }
}
