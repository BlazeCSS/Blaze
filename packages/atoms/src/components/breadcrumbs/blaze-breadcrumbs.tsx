import { Component, Element } from '@stencil/core';

@Component({
  tag: 'blaze-breadcrumbs'
})
export class Breadcrumbs {

  @Element() el: HTMLElement;

  componentDidLoad() {
    const lastCrumb = this.el.querySelector('blaze-breadcrumb:last-child .c-breadcrumbs__crumb');

    if (lastCrumb) {
      lastCrumb.setAttribute('aria-current', 'true');
    }
  }

  render() {
    return (
      <nav>
        <ol class="c-breadcrumbs">
          <slot />
        </ol>
      </nav>
    );
  }
}
