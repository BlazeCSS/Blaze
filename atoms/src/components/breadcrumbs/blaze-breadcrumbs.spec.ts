import { flush, render } from '@stencil/core/testing';
import { Breadcrumbs } from './blaze-breadcrumbs';
import { Breadcrumb } from './blaze-breadcrumb';

describe('Breadcrumbs', () => {
  it('should build', () => {
    expect(new Breadcrumbs()).toBeTruthy();
    expect(new Breadcrumb()).toBeTruthy();
  });

  let element;

  const snapIt = (name, html) => {
    it(name, async () => {
      element = await render({
        components: [Breadcrumbs, Breadcrumb],
        html
      });
      await flush(element);

      expect(element).toMatchSnapshot();
    });
  };

  snapIt(
    'nothing is rendered with no crumbs',
    '<blaze-breadcrumbs></blaze-breadcrumbs>'
  );

  snapIt(
    'one crumb',
    `<blaze-breadcrumbs>
      <blaze-breadcrumb>Home</blaze-breadcrumb>
    </blaze-breadcrumbs>`
  );

  snapIt(
    'two crumbs with a separator',
    `<blaze-breadcrumbs>
      <blaze-breadcrumb>Home</blaze-breadcrumb>
      <blaze-breadcrumb>Sub-section</blaze-breadcrumb>
    </blaze-breadcrumbs>`
  );

  snapIt(
    'linked crumb',
    `<blaze-breadcrumbs>
      <blaze-breadcrumb href="home.html">Home</blaze-breadcrumb>
      <blaze-breadcrumb href="another-page.html">Another page</blaze-breadcrumb>
      <blaze-breadcrumb>Sub-section</blaze-breadcrumb>
    </blaze-breadcrumbs>`
  );
});
