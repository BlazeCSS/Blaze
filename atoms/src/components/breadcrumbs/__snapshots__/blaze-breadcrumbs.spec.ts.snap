// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs linked crumb 1`] = `
<blaze-breadcrumbs
  class="hydrated"
>
  
      
  <blaze-breadcrumb
    class="hydrated"
    href="home.html"
  >
    Home
    <li
      class="c-breadcrumbs__crumb"
    >
      <a
        class="c-link"
        href="home.html"
      />
    </li>
  </blaze-breadcrumb>
  
      
  <blaze-breadcrumb
    class="hydrated"
    href="another-page.html"
  >
    Another page
    <li
      class="c-breadcrumbs__crumb"
    >
      <a
        class="c-link"
        href="another-page.html"
      />
    </li>
  </blaze-breadcrumb>
  
      
  <blaze-breadcrumb
    class="hydrated"
  >
    Sub-section
    <li
      class="c-breadcrumbs__crumb"
    />
  </blaze-breadcrumb>
  
    
  <ol
    class="c-text c-breadcrumbs"
  />
</blaze-breadcrumbs>
`;

exports[`Breadcrumbs nothing is rendered with no crumbs 1`] = `
<blaze-breadcrumbs
  class="hydrated"
>
  <ol
    class="c-text c-breadcrumbs"
  />
</blaze-breadcrumbs>
`;

exports[`Breadcrumbs one crumb 1`] = `
<blaze-breadcrumbs
  class="hydrated"
>
  
      
  <blaze-breadcrumb
    class="hydrated"
  >
    Home
    <li
      class="c-breadcrumbs__crumb"
    />
  </blaze-breadcrumb>
  
    
  <ol
    class="c-text c-breadcrumbs"
  />
</blaze-breadcrumbs>
`;

exports[`Breadcrumbs two crumbs with a separator 1`] = `
<blaze-breadcrumbs
  class="hydrated"
>
  
      
  <blaze-breadcrumb
    class="hydrated"
  >
    Home
    <li
      class="c-breadcrumbs__crumb"
    />
  </blaze-breadcrumb>
  
      
  <blaze-breadcrumb
    class="hydrated"
  >
    Sub-section
    <li
      class="c-breadcrumbs__crumb"
    />
  </blaze-breadcrumb>
  
    
  <ol
    class="c-text c-breadcrumbs"
  />
</blaze-breadcrumbs>
`;
